File: StriverA2Z/C++/3_ArraysQuestion/3.1_Easy/linear_search.cpp
#include<bits/stdc++.h>

using namespace std;

void linearSearch(int arr[], int n, int target) {
    bool flag = true;
    for (int i = 0; i < n; i++) {
        if (arr[i] == target) {
            flag = false;
            cout << target << " is founded at " << i;
            break;
        }
    }
    if (flag) {
        cout << "-1 not found";
    }
}

int main() {
    int arr[] = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
    linearSearch(arr, sizeof(arr) / sizeof(arr[0]), 32);
    return 0;
}

File: StriverA2Z/Java/Arrays/Easy/LinearSearch.java
public class LinearSearch {
    public static void main(String[] args) {
        int[] arr = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
        linearSearch(arr, 32);
    }

    static void linearSearch(int[] arr, int num) {
        boolean flag = true;
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] == num) {
                System.out.println(arr[i] + " founded at index " + i);
                flag = false;
                break;
            }
        }
        if (flag) {
            System.out.println("-1 not found");
        }
    }
}
